src/main/java/org/jboss/pnc/bifrost/endpoint/LogUpload.java

```diff
@@ -93,6 +102,31 @@ public String uploadFinalLog(@Valid FinalLogUpload logUpload, @Context HttpHeade
         return "ok";
     }
 
+    @Path("/{processContext}/delete")
+    @DELETE
+    @RolesAllowed("**") // FIXME change to specific allowed roles
+    public void deleteFinalLog(
+            @PathParam("processContext") String processContext,
+            // IF tag is null, then all logs are deleted
+            @QueryParam("tag") @NotBlank String tag) {
+        // parse process context
+        long processContextLong;
+        if (processContext.startsWith("build-")) {
+            processContextLong = idConverter.convert(processContext);
+        } else {
+            try {
+                processContextLong = Long.parseLong(processContext);
+            } catch (NumberFormatException e) {
+                throw new BadRequestException(
+                        "Process context " + processContext + "is not a number nor a Build process.");
+            }
+        }
+
+        log.debug("Deleting final logs with processContext: {} (long -> {})", processContext, processContextLong);
+        long deleted = FinalLog.deleteByProcessContext(processContextLong, tag, true);
+        log.debug("Deleted {} rows with processContext {} (long -> {}).", deleted, processContext, processContextLong);
```

Comment:
I wonder if we should return a bad Response object if we're trying to delete a permanent build. and a positive one if the deletion works.

Reply:
Will do!

Reply:
Is throwing an BadRequest exception valid?

Reply:
Yeah BadRequest sounds good, with a message on why this is a bad request.